src/clj/rems/api/applications_v2.clj

```diff
@@ -69,32 +80,62 @@
           :application/licenses))
 
 (mount/defstate all-applications-cache
-  :start (atom {:last-processed-event-id 0
-                :applications nil}))
+  :start (events-cache/new))
+
+(defn- group-apps-by-user [apps]
+  (->> apps
+       (mapcat (fn [app]
+                 (for [user (keys (::permissions/user-roles app))]
+                   (when-let [app (model/apply-user-permissions app user)]
+                     [user app]))))
+       (reduce (fn [apps-by-user [user app]]
+                 (update apps-by-user user conj app))
+               {})))
+
+(defn- refresh-all-applications-cache! []
+  (events-cache/refresh!
+   all-applications-cache
+   (fn [state events]
+     ;; Because enrich-with-injections is not idempotent,
+     ;; it's necessary to hold on to the "raw" applications.
```

Comment:
It does ‘(dissoc ::draft-answers ::submitted-answers ::previous-submitted-answers)’ but other than that it’s mostly idempotent. It could be made idempotent with small changes.

Reply:
moving the dissocs to `get-all-unrestricted-applications` would make sense to me

Reply:
I'm thinking that `apply-user-permissions` could do it in `hide-non-public-information`.

Reply:
that sounds good too